test/unit/sdam/server_selection.test.js

```diff
@@ -31,6 +32,67 @@ describe('server selection', function () {
     isWritablePrimary: true,
     ok: 1
   });
+  const unknown = new ServerDescription('127.0.0.1:27022', {
+    ok: 0
+  });
+
+  describe('#sameServerSelector', function () {
+    const serverDescriptions = new Map();
+    serverDescriptions.set('127.0.0.1:27017', primary);
+    serverDescriptions.set('127.0.0.1:27018', unknown);
+
+    context('when the server is unknown', function () {
+      const topologyDescription = new TopologyDescription(
+        TopologyType.ReplicaSetWithPrimary,
+        serverDescriptions,
+        'test',
+        MIN_SECONDARY_WRITE_WIRE_VERSION,
+        new ObjectId(),
+        MIN_SECONDARY_WRITE_WIRE_VERSION
+      );
+      const selector = sameServerSelector(unknown);
+      const servers = selector(topologyDescription, Array.from(serverDescriptions.values()));
+
+      it('returns an empty array', function () {
+        expect(servers).to.be.empty;
+      });
+    });
+
+    context('when the server is not unknown', function () {
```

Comment:
Could we also add a test case for when there is no server that matches the address? e.g., `selector = sameServerSelector(secondary)` ?

Reply:
Extra test case added.